tapgarden/batch.go

```diff
@@ -193,6 +194,55 @@ func (m *MintingBatch) MintingOutputKey(sibling *commitment.TapscriptPreimage) (
 	return m.mintingPubKey, m.taprootAssetScriptRoot, nil
 }
 
+// VerifyOutputScript recomputes a batch genesis output script from a batch key,
+// tapscript sibling, and set of assets. It checks multiple tap commitment
+// versions to account for legacy batches.
+func VerifyOutputScript(batchKey *btcec.PublicKey, tapSibling *chainhash.Hash,
+	genesisScript []byte, assets []*asset.Asset) (*commitment.TapCommitment,
+	error) {
+
+	// Construct a TapCommitment from the batch sprouts, and verify that the
+	// version is correct by recomputing the genesis output script.
+	tapCommitment, err := commitment.FromAssets(
```

Comment:
I think we can also use `AnchorOutputScript` here. It has a porvision to ensure certain witnesses are trimmed proeprly.

Reply:
In this case none of the assets should have witnesses because they are all genesis assets? But good point.

Reply:
Adding trimming here in case we end up using this elsewhere.